src/gConsent/manage/getAccessGrantAll.ts

```diff
@@ -149,11 +160,100 @@ async function getAccessGrantAll(
+ * Grants for the target resource, including recursive Access Grants issued over
+ * an ancestor container, the resources hierarchy is walked up to the storage root.
+ *
+ * @param resource The URL of a resource to which access grants might have been issued.
+ * @param grantShape The properties of grants to filter results.
+ * @param options Optional parameter:
+ * - `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * This can be typically used for authentication. Note that if it is omitted, and
+ * `@inrupt/solid-client-authn-browser` is in your dependencies, the default session
+ * is picked up.
+ * - `options.includeExpiredGrants`: include expired grants in the result set.
+ * @returns A void promise.
+ * @since 0.4.0
+ */
+
+async function getAccessGrantAll(
+  resource: URL | UrlString,
+  params: Partial<
+    Pick<IssueAccessRequestParameters, "access" | "purpose"> & {
+      requestor: string;
+      resource: URL | UrlString;
+    }
+  >,
+  options?: AccessBaseOptions & { includeExpired?: boolean }
+): Promise<Array<VerifiableCredential>>;
+async function getAccessGrantAll(
+  params: Partial<
+    Pick<IssueAccessRequestParameters, "access" | "purpose"> & {
+      requestor: string;
+      resource: URL | UrlString;
+    }
+  >,
+  options?: AccessBaseOptions & { includeExpired?: boolean }
+): Promise<Array<VerifiableCredential>>;
+async function getAccessGrantAll(
+  resourceOrParams:
+    | URL
+    | UrlString
+    | Partial<
+        Pick<IssueAccessRequestParameters, "access" | "purpose"> & {
+          requestor: string;
+        }
+      >,
+  paramsOrOptions:
+    | Partial<
+        Pick<IssueAccessRequestParameters, "access" | "purpose"> & {
+          requestor: string;
+        }
+      >
+    | undefined
+    | (AccessBaseOptions & { includeExpired?: boolean }),
+  options: AccessBaseOptions & { includeExpired?: boolean } = {}
+): Promise<Array<VerifiableCredential>> {
+  if (
+    typeof resourceOrParams === "string" ||
+    (resourceOrParams as URL).href !== undefined
+  ) {
+    return internal_getAccessGrantAll(
+      {
+        ...paramsOrOptions,
+        resource:
+          typeof resourceOrParams === "string"
+            ? resourceOrParams
+            : (resourceOrParams as URL).href,
+      },
+      options
+    );
+  }
+  if (typeof resourceOrParams === "object") {
+    return internal_getAccessGrantAll(
+      resourceOrParams as Partial<
+        Pick<IssueAccessRequestParameters, "access" | "purpose"> & {
+          requestor: string;
+          resource: string | URL;
+        }
+      >,
+      paramsOrOptions as AccessBaseOptions & { includeExpired?: boolean }
+    );
+  }
+  return internal_getAccessGrantAll(
+    resourceOrParams,
+    paramsOrOptions as AccessBaseOptions & { includeExpired?: boolean }
+  );
+}
```

Comment:
This could be a bit more concise: 
```
if (
    typeof resourceOrParams === "string" ||
    (resourceOrParams as URL).href !== undefined
  ) {
    return internal_getAccessGrantAll(
      {
        ...paramsOrOptions,
        resource:
          typeof resourceOrParams === "string"
            ? resourceOrParams
            : (resourceOrParams as URL).href,
      },
      options
    );
  }
  return internal_getAccessGrantAll(
    resourceOrParams as AccessParameters,
    paramsOrOptions as AccessBaseOptions & { includeExpired?: boolean }
  );
```